actors/src/main/kotlin/ktx/actors/events.kt

```diff
@@ -80,6 +80,89 @@ inline fun <Widget : Actor> Widget.onClickEvent(
+ * @return [ClickListener] instance.
+ */
+inline fun Actor.onTouchDown(crossinline listener: () -> Boolean): ClickListener {
+  val clickListener = object: ClickListener() {
+    override fun touchDown(event: InputEvent, x: Float, y: Float, pointer: Int, button: Int) = listener()
+  }
+  this.addListener(clickListener)
+  return clickListener
+}
+
+/**
+ * Attaches a [ClickListener] to this actor.
+ * @param listener invoked each time the touch of the actor is released.
+ * @return [ClickListener] instance.
+ */
+inline fun Actor.onTouchUp(crossinline listener: () -> Unit): ClickListener {
+  val clickListener = object: ClickListener() {
+    override fun touchUp(event: InputEvent, x: Float, y: Float, pointer: Int, button: Int) = listener()
+  }
+  this.addListener(clickListener)
+  return clickListener
+}
+
+/**
+ * Attaches a [ClickListener] to this actor.
+ * @param downListener invoked each time this actor is touched. Consumes the triggered [InputEvent] and the [Actor] that
+ * the listener was originally attached to. Refer to [ClickListener.touchDown] for parameter details.
+ * * @param upListener invoked each time the touch of the actor is released. Consumes the triggered [InputEvent] and the [Actor] that
+ * the listener was originally attached to. Refer to [ClickListener.touchUp] for parameter details.
+ * @return [ClickListener] instance.
+ */
+inline fun <Widget : Actor> Widget.onTouchEvent(
+  crossinline downListener: (event: InputEvent, actor: Widget) -> Boolean,
+  crossinline upListener: (event: InputEvent, actor: Widget) -> Unit): ClickListener {
+  val clickListener = object : ClickListener() {
+    override fun touchDown(event: InputEvent, x: Float, y: Float, pointer: Int, button: Int): Boolean = downListener(event, this@onTouchEvent)
+    override fun touchUp(event: InputEvent, x: Float, y: Float, pointer: Int, button: Int) = upListener(event, this@onTouchEvent)
+  }
+  this.addListener(clickListener)
+  return clickListener
+}
+
+/**
+ * Attaches a [ClickListener] to this actor.
+ * @param downListener invoked each time this actor is touched. Consumes the triggered [InputEvent] and the [Actor] that
+ * the listener was originally attached to. Refer to [ClickListener.touchDown] for parameter details.
+ * * @param upListener invoked each time the touch of the actor is released. Consumes the triggered [InputEvent] and the [Actor] that
+ * the listener was originally attached to. Refer to [ClickListener.touchUp] for parameter details.
+ * @return [ClickListener] instance.
+ */
+inline fun <Widget : Actor> Widget.onTouchEvent(
+  crossinline downListener: (event: InputEvent, actor: Widget, x: Float, y: Float) -> Boolean,
+  crossinline upListener: (event: InputEvent, actor: Widget, x: Float, y: Float) -> Unit): ClickListener {
+  val clickListener = object : ClickListener() {
+    override fun touchDown(event: InputEvent, x: Float, y: Float, pointer: Int, button: Int): Boolean = downListener(event, this@onTouchEvent, x, y)
```

Comment:
You might want to move the part after return type definition to the next line to improve readability. Same goes for other longer methods.

Reply:
Usually I use autoformat but you seem to have a different format then the "Kotlin standard" from intellij. Don't know how to take over your formatting.
But anyway, will move that to a new line :)

Reply:
I think the XML with the configuration is in the repo, but it's not fully automatic - it doesn't break some lines that it should. I was planning on adding editor config and conforming to the official formatting guide, but that would break most of the Git history.

Reply:
Ah okay, I did notice that it does "something" but not 100% perfect. But I think it is fine and I hopefully was able to fix the formatting issues.